### PR TITLE
add metrics/slis to kube-proxy health checks

### LINKED ISSUE #112959: Wire up metrics/slis for kube-proxy
### What would you like to be added?

As per, https://kep.k8s.io/3466, wire up metrics/slis for kube-proxy

### Why is this needed?

https://kep.k8s.io/3466

### PR DESCRIPTION
**Pull request by richabanker**
#### What type of PR is this?

/kind feature

#### What this PR does / why we need it:
Adds health check SLI metrics to kube-proxy. This is behind a feature gate.

#### Which issue(s) this PR fixes:

Fixes #112959

#### Does this PR introduce a user-facing change?

```release-note
If `ComponentSLIs` feature gate is enabled, then `/metrics/slis` becomes available on kube-proxy allowing you to scrape health check metrics.

```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs
- [KEP]: https://kep.k8s.io/3466
```
```
# HELP kubernetes_healthcheck [ALPHA] This metric records the result of a single healthcheck.
# TYPE kubernetes_healthcheck gauge
kubernetes_healthcheck{name="ping",type="healthz"} 1
# HELP kubernetes_healthchecks_total [ALPHA] This metric records the results of all healthcheck.
# TYPE kubernetes_healthchecks_total counter
kubernetes_healthchecks_total{name="ping",status="success",type="healthz"} 1
```

### CODE REVIEW

File: cmd/kube-proxy/app/server.go [hunk 1]
```diff
@@ -86,6 +88,10 @@ import (
 	"k8s.io/utils/pointer"
 )
 
+func init() {
+	utilruntime.Must(metricsfeatures.AddFeatureGates(utilfeature.DefaultMutableFeatureGate))
```
Comment (aojea) [inline cmd/kube-proxy/app/server.go:92]: why is this needed?

**Comment by aojea** [comment]
/assign @aojea @logicalhan 

one naive question

File: cmd/kube-proxy/app/server.go (hunk 1 above)
Comment (logicalhan) [reply cmd/kube-proxy/app/server.go:92]: You need to register the feature gates before flags are applied, so conventionally this is done through an init function.

**Review by logicalhan — APPROVED** [review]
/lgtm
/approve

File: cmd/kube-proxy/app/server.go (hunk 1 above)
Comment (aojea) [reply cmd/kube-proxy/app/server.go:92]: are we not setting them on line 257?

```
	return utilfeature.DefaultMutableFeatureGate.SetFromMap(o.config.FeatureGates)
```

Comment (logicalhan) [reply cmd/kube-proxy/app/server.go:92]: We can probably also try setting it the line before. But the init function is pretty standard way of setting this across the kubernetes code base.

Comment (Richabanker) [reply cmd/kube-proxy/app/server.go:92]: I was able to curl the /metrics/slis endpoint after commenting out the init() block- so functionally, its ok to be removed, but it makes sense to keep the init() block and follow the conventional way for initialization purposes.

Comment (logicalhan) [reply cmd/kube-proxy/app/server.go:92]: Your call @aojea 😃 we're open to either way, but I think we're slightly more inclined to follow convention.

Comment (aojea) [reply cmd/kube-proxy/app/server.go:92]: yeah, me too, I just wanted to fully understand it
Thanks for clarifying

**Comment by aojea** [comment]
/approve

clarified in https://github.com/kubernetes/kubernetes/pull/113057#discussion_r995958134

**Comment by aojea** [comment]
> Trace[132957275]: [151.860425ms] [151.860425ms] END
> panic: test timed out after 3m0s
> goroutine 32339 [running]:
> testing.(*M).startAlarm.func1()

/test pull-kubernetes-unit
